### PR TITLE
NOISSUE - Remove auth URL from SDK

### PR DESCRIPTION
**Pull request by fbugarski**
Signed-off-by: Filip Bugarski filipbugarski@gmail.com

### What does this do?
Remove authURL, and add groupsURL
### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### Have you included tests for your changes?
Not needed
### Did you document any new/modified functionality?
No

### CODE REVIEW

**Review by manuio — DISMISSED** [review]
LGTM

File: cmd/cli/main.go [hunk 1]
```diff
@@ -64,10 +64,10 @@ func main() {
 
 	// Root Flags
 	rootCmd.PersistentFlags().StringVarP(
-		&sdkConf.AuthURL,
+		&sdkConf.GroupsURL,
```
Comment (drasko) [inline cmd/cli/main.go]: Gropus is not a separate microservice, thus it should not have a configurable URL. It is either Auth (where it is today) or Things/Users (where it will be in the future).

File: pkg/sdk/go/groups.go [hunk 2]
```diff
@@ -158,51 +158,51 @@ func (sdk mfSDK) Members(groupID, token string, offset, limit uint64) (auth.Memb
...
-func (sdk mfSDK) Groups(offset, limit uint64, token string) (auth.GroupPage, error) {
-	url := fmt.Sprintf("%s/%s?offset=%d&limit=%d&tree=false", sdk.authURL, groupsEndpoint, offset, limit)
+func (sdk mfSDK) Groups(offset, limit uint64, token string) (GroupsPage, error) {
+	url := fmt.Sprintf("%s/%s?offset=%d&limit=%d&tree=false", sdk.groupsURL, groupsEndpoint, offset, limit)
 	return sdk.getGroups(token, url)
 }
 
-func (sdk mfSDK) Parents(id string, offset, limit uint64, token string) (auth.GroupPage, error) {
-	url := fmt.Sprintf("%s/%s/%s/parents?offset=%d&limit=%d&tree=false&level=%d", sdk.authURL, groupsEndpoint, id, offset, limit, auth.MaxLevel)
+func (sdk mfSDK) Parents(id string, offset, limit uint64, token string) (GroupsPage, error) {
```
Comment (drasko) [inline pkg/sdk/go/groups.go:164]: Using a Groups page, as defined in SDK, is OK (i.e. we redefined structs apparently in SDK for Users, Things, Groups, etc...)

**Review by drasko — CHANGES_REQUESTED** [review]

File: cmd/cli/main.go [hunk 3]
```diff
@@ -17,7 +17,6 @@ const defURL string = "http://localhost"
 func main() {
 	msgContentType := string(sdk.CTJSONSenML)
 	sdkConf := sdk.Config{
-		AuthURL:         defURL,
```
Comment (drasko) [inline cmd/cli/main.go]: Why is this removed? CLI does not call Auth ever? Even for handlig Groups?

File: pkg/sdk/go/groups.go [hunk 4]
```diff
@@ -11,17 +11,25 @@ import (
...
+	MaxLevel       = uint64(5)
+	MinLevel       = uint64(1)
+)
 
 type assignRequest struct {
 	Type    string   `json:"type,omitempty"`
 	Members []string `json:"members"`
 }
 
+type Member struct {
```
Comment (drasko) [inline pkg/sdk/go/groups.go]: I am not sure, but I think these kind of hig-level struct types are defined elsewhere. Probably in the `sdk.go`.

File: pkg/sdk/go/groups.go [hunk 5]
```diff
@@ -11,17 +11,25 @@ import (
...
 )
 
-const groupsEndpoint = "groups"
+const (
+	groupsEndpoint = "groups"
+	MaxLevel       = uint64(5)
+	MinLevel       = uint64(1)
+)
 
 type assignRequest struct {
```
Comment (drasko) [inline pkg/sdk/go/groups.go]: Isn't there `requests.go`? Are you shure that this should be here?

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]
